_1327/static/js/utils.js

```diff
@@ -48,3 +48,24 @@ var debounce = function (func, threshold) {
 		}, threshold);
 	};
 };
+
+
+// a nice splice function taken from here:
+// https://stackoverflow.com/questions/4313841/insert-a-string-at-a-specific-index
+if (!String.prototype.splice) {
+    /**
+     * {JSDoc}
+     *
+     * The splice() method changes the content of a string by removing a range of
+     * characters and/or adding new characters.
+     *
+     * @this {String}
+     * @param {number} start Index at which to start changing the string.
+     * @param {number} delCount An integer indicating the number of old chars to remove.
+     * @param {string} newSubStr The String that is spliced in.
+     * @return {string} A new string with the spliced substring.
+     */
+    String.prototype.splice = function(start, delCount, newSubStr) {
```

Comment:
this is used in only one place and `delCount` is `0`, so it looks like too many lines of code (or rather documentation) for what's happening. why not simply use the substring from the beginning to the cursor position, then the added text and finally the substring from the cursor position to the end in `documents_edit.html`?